### PR TITLE
Enhance filter variables to support arrays

### PR DESCRIPTION
**Pull request by billwashere**
This pull request checks if the filter type is an array and uses the _in operator instead.

### CODE REVIEW

File: src/buildVariables.js [hunk 1]
```diff
@@ -24,7 +24,10 @@ const buildGetListVariables = introspectionResults => (
         let filter;
         if (key === 'ids') {
             filter = { id: { _in: filterObj['ids'] } };
-        } else {
+        
+	} else if(Array.isArray(filterObj[key])) {
+		filter = { [key]: { _in: filterObj[key] } };
+	} else {
```
Comment (webdeb) [inline src/buildVariables.js]: makes sense, but please keep the code style consistent

**Comment by cpursley** [comment]
This looks useful @billwashere ~ what do you think @Steams ?

**Comment by cpursley** [comment]
@billwashere ~ How do you add an Array filter type? Couldn't find it in the react-admin docs.

**Comment by billwashere** [comment]
Technically the usage is data provider independent.but by passing a filter prop with an array rather than a number or string
Ie
`
{ "filter_column": [1,2] }
`
Example:
`
<ReferenceInput
     source="post_id"
     reference="posts"
     filter={{ author_id: [1,2] }}>
    <SelectInput optionText="title" />
</ReferenceInput>
`

**Pull request merged**